src/Orleans/Orleans.csproj

```diff
@@ -1,522 +1,92 @@
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(BuildFlavor)' == 'Current' ">
+    <PackageReference Include="System.ComponentModel.TypeConverter" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Diagnostics.FileVersionInfo" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Diagnostics.Process" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Diagnostics.TraceSource" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Linq.Expressions" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Net.NameResolution" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Net.NetworkInformation" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Reflection.Emit" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Reflection.Emit.Lightweight" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Reflection.TypeExtensions" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Runtime" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Runtime.Serialization.Formatters" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Runtime.Serialization.Primitives" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Security.Cryptography.Algorithms" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Threading.Thread" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Threading.ThreadPool" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Xml.XmlDocument" Version="$(SystemRuntimeVersion)" />
+    <PackageReference Include="System.Xml.XPath.XmlDocument" Version="$(SystemRuntimeVersion)" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <OrleansBootstrapSteps>
+      OrleansDllBootstrapUsingCodeGen
+    </OrleansBootstrapSteps>
+  </PropertyGroup>
+
+  <Target Name="OrleansBootstrap" BeforeTargets="AssignTargetPaths" DependsOnTargets="$(OrleansBootstrapSteps)" Condition="'$(Bootstrap)' != 'true'" />
+
+  <Target Name="OrleansDllBootstrapUsingCodeGen" Inputs="@(ReferencePath)" Outputs="$(IntermediateOutputPath)$(TargetName).orleans.g.cs" Condition="'$(Bootstrap)' != 'true'">
+    <Message Text="[OrleansDllBootstrapUsingCodeGen] - Building ClientGenerator start" Importance="high" />
     <PropertyGroup>
-      <BootstrapOutputPath>$(OutDir)Bootstrap\</BootstrapOutputPath>
-      <BootstrapOutputPath Condition="'$(BuildingInsideVisualStudio)' == 'true'">$(ProjectDir)$(OutDir)Bootstrap\</BootstrapOutputPath>
-      <ExcludeCodeGen>$(DefineConstants);EXCLUDE_CODEGEN</ExcludeCodeGen>
+      <!-- Visual Studio or MsBuild .sln build -->
+      <BootstrapOutputPath>$(SourceRoot)/Bootstrap/$(BuildFlavor)/$(Configuration)/</BootstrapOutputPath>
```

Comment:
This is where to build bootstrap there where to use it from. Different purpose, but pointing to the same directory now.